ext4: return error when calling ext4_ext_split failed

From: Shen Feng <shen@cn.fujitsu.com>


ext4_ext_create_new_leaf must return error when its
calling to ext4_ext_split failed.

Signed-off-by: Shen Feng <shen@cn.fujitsu.com>
Signed-off-by: Mingming Cao <cmm@us.ibm.com>
Reviewed-by: Aneesh Kumar K.V <aneesh.kumar@linux.vnet.ibm.com>
---
 fs/ext4/extents.c |    2 ++
 1 file changed, 2 insertions(+)

Index: linux-2.6.26-rc5-git6/fs/ext4/extents.c
===================================================================
--- linux-2.6.26-rc5-git6.orig/fs/ext4/extents.c	2008-06-12 15:09:19.000000000 -0700
+++ linux-2.6.26-rc5-git6/fs/ext4/extents.c	2008-06-12 15:15:00.000000000 -0700
@@ -981,6 +981,8 @@ repeat:
 		/* if we found index with free entry, then use that
 		 * entry: create all needed subtree and add new leaf */
 		err = ext4_ext_split(handle, inode, path, newext, i);
+		if (err)
+			goto out;
 
 		/* refill path */
 		ext4_ext_drop_refs(path);
